### PR TITLE
Add checkpoint optionnal arg in train tool. Allow to retrain or fine tune.

### PR DESCRIPTION
**Pull request by ocourtin**

### CODE REVIEW

**Comment by ocourtin** [comment]
Hi Daniel,

I did't see 553c579#diff-2c784d9a2df2a668d53096d329987959 before...
Thanks to point this out !

A typical use case on my own, is:
- Train a model from a generic dataset
- Then retrain it, on a more specific dataset (e.g different imagery provider)

So it's not exactly the same use case then the previous 'resume'.
It's more close to the ResNet50/ImageNet weights reuse in the UNet encoder.

So it's why, for now, i did not take in consideration the learning rate decay, in the PR.

Let me know, if you still have any questions or concerns about it.

**Comment by ocourtin** [comment]
And keep thinking about it, could be interesting to have both use cases indeed, 
dunno what could be the more elegant user side:

- two differents args: resume and retrain 
- resume arge, and if not able to retrieve the epoch from filename, start at 0
- resume arg, and resume_epoch arg

My preference is the last one.

And i could provide a related new PR.

**Comment by daniel-j-h** [comment]
If your current use case is for transfer learning I'm fine with supporting only that as a first step :+1:

**Comment by daniel-j-h** [comment]
I almost forgot: you may want to try https://github.com/mapbox/robosat/issues/72 for fine-tuning or at least use a small(er) learning rate.
